src/packagedcode/npm.py

```diff
@@ -575,10 +576,16 @@ def dist_mapper(dist, package):
     integrity = dist.get('integrity') or None
     if integrity:
         algo, _, b64value = integrity.partition('-')
-        assert 'sha512' == algo
         algo = algo.lower()
-        sha512 = b64value.decode('base64').encode('hex')
+        assert 'sha512' == algo
+
+        decoded_b64value = base64.b64decode(b64value)
+        if type(decoded_b64value) == str:
+            sha512 = decoded_b64value.encode('hex')
+        elif type(decoded_b64value) == bytes:
```

Comment:
That may not work between Py2 and Py3.
You may instead try this:
```
        if isinstance(decoded_b64value, string_types):
            sha512 = decoded_b64value.encode('hex')
        elif isinstance(decoded_b64value, binary_type):
....
```

Reply:
Note that I think that b64decode results may always be bytes on Python 2 or Python 3 so you may not need the type test

Reply:
I made this change in response to test failures on the python2 side. The latest code now works on both py3 and py2. `b64decode` returns bytes on python3 and string in python2, hence my code above. 

Unless I am missing something, why would `isinstance` be preferable here?

Reply:
> Unless I am missing something, why would isinstance be preferable here?

that's catching subclasses too and I use it as an automatism but that's not the main point. Rather using `six.string_types` and `six.binary_types` is what matters more IMHO for portability.